### ISSUE TITLE
Incorrect UPPER_SNAKE_CASE hint with short constant names

### ISSUE THREAD

**Issue by quest-sage** [rust-analyzer/rust-analyzer#6397]
![image](https://user-images.githubusercontent.com/71211243/97486520-745a7f80-1953-11eb-8b49-0f652ffad664.png)
The image says it all really. I'm using the current most recent version of non-nightly rust-analyzer (updated about an hour ago when I launched VS Code, not sure exactly which version it is).

I have found that the hint incorrectly shows up if and only if there are no underscores between two non-underscore characters. E.g. `AMOUNT_` shows the hint (it says I should write `A_M_O_U_N_T_`), but `AMOUNT_2` does not.

**Comment by lnicola** [comment]
Can you try in nightly? I think it's fixed:

![image](https://user-images.githubusercontent.com/308347/97487063-06b74f00-1965-11eb-978f-40928f4839a9.png)

**Comment by ArifRoktim** [comment]
^ Yeah, it's fixed in nightly. This is a duplicate of #6345, fixed in #6350, which isn't part of the latest release.

If you want, you could add
```json
"rust-analyzer.diagnostics.disabled": ["incorrect-ident-case"],
```
to your configuration until the next release.

**Comment by quest-sage** [comment]
Thanks, I searched the issue tracker for duplicates but I must have been searching `is:open`. Thanks for your support!
